src/url-state-machine.js

```diff
@@ -1419,66 +1107,86 @@ function urlToUnicode(domain) {
   }
 }
 
-function init(url, base) {
-  /*jshint validthis:true */
-  if (this === undefined) {
-    throw new TypeError("Failed to construct 'URL': Please use the 'new' operator, " +
-      "this DOM object constructor cannot be called as a function.");
-  }
-  if (arguments.length === 0) {
-    throw new TypeError("Failed to construct 'URL': 1 argument required, but only 0 present.");
-  }
+module.exports.createURLConstructor = function () {
+  // TODO
```

Comment:
Hmm actually I think this is legacy; createURLConstructor is no longer necessary since we just export the URL constructor directly.